src/functions/Output.ps1

```diff
@@ -477,6 +477,83 @@ function ConvertTo-HumanTime {
     }
 }
 
+function Get-SkipRemainingOnFailurePlugin ($SkipRemainingOnFailure) {
+    $p = @{
+        Name = "SkipRemainingOnFailure"
+    }
+
+    if ($SkipRemainingOnFailure -eq 'Block') {
+        $p.EachTestTeardownEnd = {
+            param($Context)
+
+            # If test is not marked skipped and failed
+            # Go through block tests and child tests and mark unexecuted tests as skipped
+            if (-not $Context.Test.Skipped -and -not $Context.Test.Passed) {
+
+                foreach ($test in $Context.Block.Tests) {
+                    if (-not $test.Executed) {
+                        $test.Skipped = $true
+                    }
+                }
+
+                foreach ($test in ($Context.Block | View-Flat)) {
+                    if (-not $test.Executed) {
+                        $test.Skipped = $true
+                    }
+                }
```

Comment:
Setting `Skipped = $true` will report skipped, but won't actually stop the tests and setups from being executed. Try `$test.Skip = $true`.

Reply:
Ah yeah was going to ask about this. I saw both while debugging, make sense to have one for reporting and one for setting test in skipped state 👍